sql/sql_zip_dict.h

```diff
@@ -19,9 +19,35 @@ this program; if not, write to the Free Software Foundation, Inc.,
 #ifndef SQL_ZIP_DICT_INCLUDED
 #define SQL_ZIP_DICT_INCLUDED
 
+#include "lex_string.h"
 #include "my_inttypes.h"
+#include "sql/dd/types/table.h"  // dd::enum_column_types
+#include "sql/mdl.h"
 
 class THD;
+struct TABLE_SHARE;
+
+namespace compression_dict {
+
+const char *const dict_table_str =
```

Comment:
Defining (not declaring) in header directly, doesn't this result in multiple definitions per including source file - put extern declarations here, and definitions in .cc?

Reply:
It doesn't result in multiple definition error because of const * ptr const has 'internal' linkage.

```
    [C++11: 3.5/3]: A name having namespace scope (3.3.6) has internal linkage if it is the name of

        a variable, function or function template that is explicitly declared static; or,
        a variable that is explicitly declared const or constexpr and neither explicitly declared extern nor previously declared to have external linkage; or
        a data member of an anonymous union.
```
I generally do what you have said. For a change, I tried different thing which is also correct.

if you prefer deinition in .cc file, let me know.

Reply:
If I understand correctly, internal linkage here will result in one copy per including source file?